### ISSUE TITLE
multithreading in `scipy.fft` disabled on Windows

### ISSUE THREAD

**Issue by rgommers** [scipy/scipy#16352]
This is an issue that I just found again. We haven't fixed it, so this will be a performance regression for `scipy.fft` functions on Windows for 1.9.0 in wheels and conda-forge packages (it's gated on Mingw-w64 usage, so the conda packages in the `defaults` channel or wheels from https://www.lfd.uci.edu/~gohlke/pythonlibs/#scipy will not be affected).

Usage of `pthreads` is disabled here: https://github.com/scipy/scipy/blob/main/scipy/fft/_pocketfft/meson.build#L13-L17

The in-progress fix at https://github.com/mreineck/pocketfft/pull/3 does not yet work.

**Comment by mreineck** [comment]
I'm sorry that I don't have positive news on this ... since I neither have access to Windows nor the required expertise with MINGW, I can't take an active role.

**Comment by rgommers** [comment]
@peterbell10 and I discussed this, and agreed that a fix is nice to have but not a blocker for 1.9.0, because it is only a performance issue and not a correctness one.

**Comment by matthew-brett** [comment]
@mreineck - can I help with setting up Windows access somehow?   I've done that on various machines, virtual remote, virtual local, and via dual boot (on an Intel Mac).

**Comment by mreineck** [comment]
Unfortnately I don't think this would help much ... the main problem is not that I don't have access to Windows, but rather that I have no experience with it. The issue is not really a bug in the FFT code but some low-level quirk in MinGW, and I'm no use at all when it comes to locating and fixing (or working around) this :-/

**Comment by dcmvdbekerom** [comment]
Is there any update on this issue?
When I set workers via `set_workers()` and read them through `get_workers()` it always returns 1, I guess this means multithreading is disabled on my system..
Would rebuilding locally with MSVC help?

**Comment by rgommers** [comment]
Yes, rebuilding with MSVC will give you multi-threaded behavior. You'll need a matching Fortran compiler though - Intel Fortran works best.

No other updates on this issue.

**Comment by carlkl** [comment]
See some news: https://github.com/msys2/MINGW-packages/issues/10164#issuecomment-5343181064 and https://github.com/msys2/MINGW-packages/issues/10164#issuecomment-5343576953.

**Comment by carlkl** [comment]
My question is better hosted on this issue: @rgommers: is this issue still relevant for vanilla scipy on windows? What are the plans for the standard scipy windows toolchains today? Is there a plan to use a mingw-w64 toolchain in future?

**Comment by rgommers** [comment]
This shouldn't be relevant anymore, we're building with `clang-cl` on Windows now. And while we of course want mingw-w64 to work reliably, I don't see us going back to building wheels or conda-forge packages with it. With Fortran out of the picture, there's no reason to.

**Comment by carlkl** [comment]
I have no further objections.

**Comment by rgommers** [comment]
Okay, let's close this then as sufficiently addressed. Thanks @carlkl & all!
